tests/libvmi/vmi.go

```diff
@@ -107,22 +108,33 @@ func WithNodeSelectorFor(node *k8sv1.Node) Option {
 // WithUefi configures EFI bootloader and SecureBoot.
 func WithUefi(secureBoot bool) Option {
 	return func(vmi *v1.VirtualMachineInstance) {
-		vmi.Spec.Domain.Firmware = &v1.Firmware{
-			Bootloader: &v1.Bootloader{
-				EFI: &v1.EFI{
-					SecureBoot: pointer.BoolPtr(secureBoot),
-				},
-			},
+		if vmi.Spec.Domain.Firmware == nil {
+			vmi.Spec.Domain.Firmware = &v1.Firmware{}
 		}
+		if vmi.Spec.Domain.Firmware.Bootloader == nil {
+			vmi.Spec.Domain.Firmware.Bootloader = &v1.Bootloader{}
+		}
+		if vmi.Spec.Domain.Firmware.Bootloader.EFI == nil {
+			vmi.Spec.Domain.Firmware.Bootloader.EFI = &v1.EFI{}
+		}
+		if secureBoot {
+			// secureBoot Requires SMM to be enabled
+			vmi.Spec.Domain.Firmware.Bootloader.EFI.SecureBoot = pointer.Bool(secureBoot)
+			vmi.Spec.Domain.Features.SMM.Enabled = pointer.Bool(secureBoot)
+		} else {
+			vmi.Spec.Domain.Firmware.Bootloader.EFI.SecureBoot = pointer.Bool(secureBoot)
+		}
```

Comment:
The assignment to `EFI.SecureBoot` is duplicated. Maybe go with:
```go
vmi.Spec.Domain.Firmware.Bootloader.EFI.SecureBoot = pointer.Bool(secureBoot)
if secureBoot {
	vmi.Spec.Domain.Features.SMM.Enabled = pointer.Bool(secureBoot)
}
```

Reply:
Yes, I agree as well.

But let me ask this: Why this logic needs to be placed in here (that SMM needs to be enabled as a side effect of setting the secure boot)?
This is up to caller that construct the VMI spec. Maybe the test wants to check what happens when there are not set correctly?

Generally speaking, the building options should be very simple and with as little logic as possible.
Once I see such a side effect, I am automatically suspicious that too much is assumed.

Reply:
Hey ,
While i agree that the building option should be very simple that calling `WithUefi` function

Reply:
Hey ,
While i agree that the building option should be very simple i think that calling `WithUefi` function with `SecureBoot=true` 
and find out that the the secureBoot isn't enable will cause More harm than good and mislead.

Reply:
I tend to agree with Barak on that one.
If secure boot is dependent on SMM and cannot be enabled without it, then `WithUEFI(secureBoot=true)` should enable it IMO.

> Maybe the test wants to check what happens when there are not set correctly?

That's a corner case. IMO, in that case, the spec needs to be defined manually without libvmi. Again, I don't think that libvmi should be dedicated for tests only.